### DISCUSSION TITLE
.LayoutUpdated fired although FrameworkElement no longer is part of the visual tree?!? (not an UNO issue though)

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#18825] [Q&A]
Well, just another day where I learnt something new ... I guess.

So far may layman thinking was, that FrameworkElement events would not be fired after the element was removed from the visual tree, in particular .[LayoutUpdated ](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.frameworkelement.layoutupdated?view=windows-app-sdk-1.6&devlangs=csharp&f1url=%3FappId%3DDev17IDEF1%26l%3DEN-US%26k%3Dk(Microsoft.UI.Xaml.FrameworkElement.LayoutUpdated)%3Bk(DevLang-csharp)%26rd%3Dtrue)would not be fired

Appears, my understanding was wrong. Please check out attached repo and run the WinUI3 head:
- click the button to remove it from the visual tree
- then place a breakpoint in the `MainPage_LayoutUpdated `method and resize the app window -> `.LayoutUpdated` still is fired

[Minimal.9.zip](https://github.com/user-attachments/files/17800901/Minimal.9.zip)

This also leaves the question if the button eventually would be garbage collected at all or not?!?

Now I wonder, if I'm doing something wrong with my repo or if my understand was incomplete so far. Any feedback would be appreciated.

**Comment by jeromelaban** [comment]
Thanks for the report. As far as I can tell, some of those events may still be raised even if the element is not in the tree, particularly if the event was scheduled to be raised when the item was still in the tree.

You may want to handle those scenarios using the `IsLoaded` property.

**Reply by DierkDroth** [reply]
Thanks for getting back to me on that one @jeromelaban. I'm actually using .LayoutUpdated since a while back we both had a conversation about a lifecycle management issue where `.Loading` and/or `.Loaded` failed, and you pointed me to `.LayoutUpdated`, which worked around the UNO WASM 'limitation' - and still does. So, I'm using it only in case 'where I really needed it'...

> particularly if the event was scheduled to be raised when the item was still in the tree.

FYI in my actual repo the event is triggered although the element for sure no longer is part of the visual tree, but still...

**Reply by MartinZikmund** [reply]
@DierkDroth this is actually the correct behavior - `LayoutUpdated` is triggered on all elements associated with the UI thread - see more info in https://github.com/unoplatform/uno/issues/7002

**Reply by DierkDroth** [reply]
@MartinZikmund thanks for your feedback. To be clear: I'm observing this on WinUI3 (native) - I didn't even check UNO WASM.

> LayoutUpdated is triggered on all elements associated with the UI thread 

Curious: where did you find that information?
